modules/preview-react/pill/lib/Pill.tsx

```diff
@@ -0,0 +1,219 @@
+    },
+    focus: {
+      icon: colors.licorice500,
+      background: colors.soap300,
+      border: colors.blueberry400,
+      focusRing: focusRing({
+        width: 1,
+      }),
+    },
+    disabled: {
+      icon: colors.licorice100,
+      label: colors.licorice100,
+      background: colors.soap100,
+      border: colors.licorice100,
+      opacity: '1',
+    },
+  };
+};
+
+const getRemovablePillColors = (disabled?: boolean) => {
+  return {
+    default: {
+      background: colors.soap300,
+      icon: colors.licorice200,
+      label: colors.blackPepper400,
+      border: disabled ? colors.licorice100 : colors.licorice200,
+    },
+
+    hover: {
+      icon: disabled ? colors.licorice100 : colors.licorice500,
+      background: disabled ? colors.soap100 : colors.soap300,
+      border: disabled ? colors.licorice100 : colors.licorice200,
+      label: disabled ? colors.licorice100 : colors.blackPepper400,
+    },
+    active: {
+      icon: disabled ? colors.licorice100 : colors.licorice500,
+      background: disabled ? colors.soap100 : colors.soap300,
+      border: disabled ? colors.licorice100 : colors.licorice500,
+      label: disabled ? colors.licorice100 : colors.blackPepper400,
+    },
+    focus: {
+      icon: colors.licorice200,
+      background: colors.soap300,
+      label: colors.blackPepper400,
+      focusRing: focusRing({width: 0, innerColor: 'transparent', outerColor: 'transparent'}),
+    },
+    disabled: {},
+  };
+};
+
+const pillBaseStyles: CSSObject = {
+  display: 'inline-flex',
+  alignItems: 'center',
+  borderRadius: borderRadius.m,
+  flexShrink: 0,
+  ...type.levels.subtext.large,
+  boxShadow: 'none',
+  outline: 'none',
+  fontWeight: 500,
+  WebkitFontSmoothing: 'antialiased',
+  MozOsxFontSmoothing: 'grayscale',
+  width: 'fit-content',
+  padding: `2px ${space.xxs}`,
+  height: space.m,
+  position: 'relative',
+};
+
+const StyledBasePill = styled(BaseButton.as('button'))<StyledType & PillProps>(
+  {
+    ...pillBaseStyles,
+
+    '&:active, &active:hover': {
+      'span[data-count="ck-pill-count"]': {
+        backgroundColor: colors.soap600,
+      },
+    },
+  },
+  boxStyleFn
+);
+
+const StyledNonInteractivePill = styled(StyledBasePill)({
+  cursor: 'default',
+  overflow: 'revert', // override BaseButton overflow styles so the click target exists outside the pill for removable
+});
+
+export const Pill = createComponent('button')({
+  displayName: 'Pill',
+  Component: ({children, model, ...config}: PillProps, ref, Element) => {
+    const value = useDefaultModel(model, config, usePillModel);
+    const {onClick, onDelete, maxWidth, ...elemProps} = config;
+
+    const actualEl = useConstant(() => {
+      if (Element === 'button') {
+        return !!onDelete || !onClick ? 'span' : 'button';
+      }
+      return Element;
+    });
+    return (
+      <PillModelContext.Provider value={value}>
+        <>
+          {!config.onClick && !config.onDelete && (
```

Comment:
`onClick` && `onDelete` renders nothing.... hmmmm

Reply:
Update on this: We allow the consumer to to do this and it does render something, but it renders a pill as an `interactive` element. The `onClick` takes precedence. If they provide both click event PLUS an `Pill.IconButton` they get two warnings. One of them being reacts warning of rendering a button within a button. The other is ours saying: `It looks like you've provided two different click events.Please provide either an onClick OR an onDelete on the <Pill/> component. If both are provided, onClick will take precedence.`

I think this a slightly better experience than a potentially hidden console.warn and the consumer missing something because something isn't rendering.

CC: @alanbsmith @NicholasBoll